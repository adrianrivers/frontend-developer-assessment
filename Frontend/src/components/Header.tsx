import React from 'react'
import { Container, Box, Image, Center, Link } from '@chakra-ui/react'

export default function Header(): JSX.Element {
  return (
    <Container as="header">
      <Box height={16} marginY={16}>
        <Center>
          <Link href="https://clearpoint.digital" target="_blank">
            <Image src="/clearPointLogo.png" width="200px" />
          </Link>
        </Center>
      </Box>
    </Container>
  )
}
