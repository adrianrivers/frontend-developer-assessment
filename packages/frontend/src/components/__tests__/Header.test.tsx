import { render } from '@testing-library/react'
import React from 'react'
import Header from '../Header'

describe('Header', () => {
  it('should render a Header component', () => {
    const { container } = render(<Header />)
    const img = container.querySelector('img') as HTMLImageElement
    const link = img.parentElement as HTMLAnchorElement

    expect(img.src).toContain('/clearPointLogo.png')
    expect(link.href).toBe('https://clearpoint.digital/')
    expect(container).toMatchSnapshot()
  })
})
