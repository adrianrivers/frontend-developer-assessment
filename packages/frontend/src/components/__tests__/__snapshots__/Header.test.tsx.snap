// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render a Header component 1`] = `
<div>
  <header
    class="chakra-container css-0"
  >
    <div
      class="css-alqwbq"
    >
      <div
        class="css-gmuwbf"
      >
        <a
          class="chakra-link css-0"
          href="https://clearpoint.digital"
          target="_blank"
        >
          <img
            class="chakra-image css-1domaf0"
            src="/clearPointLogo.png"
          />
        </a>
      </div>
    </div>
  </header>
</div>
`;
